oracle/src/test/scala/zio/sql/oracle/OracleSqlModuleSpec.scala

```diff
@@ -57,6 +64,195 @@ object OracleSqlModuleSpec extends OracleRunnableSpec with ShopSchema {
+          UUID.randomUUID(),
+          LocalDate.of(2022, 1, 1)
+        ),
+        (
+          UUID.randomUUID(),
+          UUID.randomUUID(),
+          LocalDate.of(2022, 1, 5)
+        )
+      )
+
+      val command = insertInto(orders)(
+        orderId,
+        fkCustomerId,
+        orderDate
+      ).values(rows)
+
+      assertZIO(execute(command))(equalTo(2))
+    },
+    test("Can insert all supported types") {
+      val sqlMinDateTime = LocalDateTime.of(-4713, 1, 1, 0, 0)
+      val sqlMaxDateTime = LocalDateTime.of(9999, 12, 31, 23, 59)
+
+      val sqlInstant =
+        Gen.instant(sqlMinDateTime.toInstant(ZoneOffset.MIN), sqlMaxDateTime.toInstant(ZoneOffset.MAX))
+
+      val sqlYear = Gen.int(-4713, 9999).filter(_ != 0).map(Year.of)
+
+      val sqlLocalDate = for {
+        year  <- sqlYear
+        month <- Gen.int(1, 12)
+        maxLen = if (!year.isLeap && month == 2) 28 else Month.of(month).maxLength
+        day   <- Gen.int(1, maxLen)
+      } yield LocalDate.of(year.getValue, month, day)
+
+      val sqlYearMonth = for {
+        year  <- sqlYear
+        month <- Gen.int(1, 12)
+      } yield YearMonth.of(year.getValue(), month)
+
+      val sqlLocalDateTime =
+        Gen.localDateTime(sqlMinDateTime, sqlMaxDateTime)
+
+      val sqlZonedDateTime = for {
+        dateTime <- sqlLocalDateTime
+        zoneId   <- Gen.zoneId
+      } yield ZonedDateTime.of(dateTime, zoneId)
+
+      val sqlOffsetTime =
+        Gen.offsetTime.filter(_.getOffset.getTotalSeconds % 60 == 0)
+
+      val sqlOffsetDateTime =
+        Gen
+          .offsetDateTime(sqlMinDateTime.atOffset(ZoneOffset.MIN), sqlMaxDateTime.atOffset(ZoneOffset.MAX))
+          .filter(_.getOffset.getTotalSeconds % 60 == 0)
+
+      val gen = (
+        Gen.uuid,
+        Gen.chunkOf(Gen.byte),
+        Gen.bigDecimal(Long.MinValue, Long.MaxValue),
+        Gen.boolean,
+        Gen.char,
+        Gen.double,
+        Gen.float,
+        sqlInstant,
+        Gen.int,
+        Gen.option(Gen.int),
+        sqlLocalDate,
+        sqlLocalDateTime,
+        Gen.localTime,
+        Gen.long,
+        sqlOffsetDateTime,
+        sqlOffsetTime,
+        Gen.short,
+        Gen.string,
+        Gen.uuid,
+        sqlZonedDateTime,
+        sqlYearMonth,
+        Gen.finiteDuration
+      ).tupleN
+      check(gen) { row =>
+        val insert = insertInto(allTypes)(
+          id,
+          bytearrayCol,
+          bigdecimalCol,
+          booleanCol,
+          charCol,
+          doubleCol,
+          floatCol,
+          instantCol,
+          intCol,
+          optionalIntCol,
+          localdateCol,
+          localdatetimeCol,
+          localtimeCol,
+          longCol,
+          offsetdatetimeCol,
+          offsettimeCol,
+          shortCol,
+          stringCol,
+          uuidCol,
+          zonedDatetimeCol,
+          yearMonthCol,
+          durationCol
+        ).values(row)
+
+        // TODO: ensure we can read values back correctly
```

Comment:
has this worked? test like this would be awesome!

Reply:
Hi,
if I remember correctly there was some issue with reading the data back. Can't check right now because of m1 chip.
Long term we should definitely enable this test.